src/middle/ir_writer.rs

```diff
@@ -254,7 +254,7 @@ impl IRWriter {
                                  if !acc.is_empty() {
                                      format!("{}, {}", acc, x)
                                  } else {
-                                     format!("{}", x)
+                                     x
```

Comment:
Yep
`x` is a `std::string::String` here. It's like calling `puts(x)` instead of `printf("%s\n", x)` in c (but much safer and to the buffer).